source/extensions/http/cache/file_system_http_cache/cache_file_fixed_block.h

```diff
@@ -0,0 +1,162 @@
+   * bodySize serializes to an 8-byte uint.
+   * @return the size in bytes.
+   */
+  static constexpr size_t size() { return sizeof(uint32_t) * 4 + sizeof(uint64_t); }
+
+  /**
+   * fileId is a compile-time fixed value used to identify that this is a cache file.
+   * @return the file ID.
+   */
+  std::array<char, 4> fileId() const { return file_id_; }
+
+  /**
+   * cacheVersionId is a compile-time fixed value that should be consistent between
+   * versions of the file cache implementation. Changing version in code will
+   * invalidate all cache entries where the version ID does not match.
+   * @return the cache version ID.
+   */
+  std::array<char, 4> cacheVersionId() const { return cache_version_id_; }
+
+  /**
+   * the size of the serialized proto message capturing headers and metadata.
+   * @return the size in bytes.
+   */
+  size_t headerSize() const { return header_size_; }
+
+  /**
+   * the size of the serialized proto message capturing trailers.
+   * @return the size in bytes.
+   */
+  size_t trailerSize() const { return trailer_size_; }
+
+  /**
+   * the size of the http body of the cache entry.
+   * @return the size in bytes.
+   */
+  size_t bodySize() const { return body_size_; }
+
+  /**
+   * sets the size of the serialized http headers, plus key and metadata, in the header block.
+   * @param sz The size of the serialized headers, key and metadata.
+   */
+  void setHeadersSize(size_t sz) { header_size_ = sz; }
+
+  /**
+   * sets the size of the serialized trailers in the header block.
+   * @param sz The size of the serialized trailers.
+   */
+  void setTrailersSize(size_t sz) { trailer_size_ = sz; }
+
+  /**
+   * sets the size of the serialized body in the header block.
+   * @param sz The size of the body data.
+   */
+  void setBodySize(size_t sz) { body_size_ = sz; }
+
+  /**
+   * the offset from the start of the file to the start of the serialized headers proto.
+   * @return the offset in bytes.
+   */
+  static off_t offsetToHeaders() { return size(); }
+
+  /**
+   * the offset from the start of the file to the start of the body data.
+   * @return the offset in bytes.
+   */
+  off_t offsetToBody() const { return offsetToHeaders() + headerSize(); }
+
+  /**
+   * the offset from the start of the file to the start of the serialized trailers proto.
+   * @return the offset in bytes.
+   */
+  off_t offsetToTrailers() const { return offsetToBody() + bodySize(); }
+
+  /**
+   * is this a valid cache file header block for the current code version?
+   * @return True if the block's cache version id and file id match the current version.
+   */
+  bool isValid() const;
+
+private:
+  /**
+   * sets the fileId, a fixed value used to identify that this is a cache file.
+   * This is private so only the unit test can call it, as the file ID should be
+   * a compile-time constant in ordinary use.
+   * @param id The file ID to set.
+   */
+  void setFileId(std::array<char, 4> id) { file_id_ = id; }
```

Comment:
Rather than modifying these logically constant values for tests, I think it would be cleaner and more representative of reality to have a test just modify the appropriate bytes of a file, and verify that the file gets rejected.

Reply:
Good call. Done.